org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/editor/folding/XtextFoldingTest.xtend

```diff
@@ -0,0 +1,242 @@
+/*******************************************************************************
+ * Copyright (c) 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.xtext.ui.editor.folding
+
+import org.eclipse.xtext.testing.InjectWith
+import org.eclipse.xtext.testing.XtextRunner
+import org.eclipse.xtext.ui.testing.AbstractFoldingTest
+import org.eclipse.xtext.xtext.ui.XtextUiInjectorProvider
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * @author miklossy - Initial contribution and API
+ */
+@RunWith(XtextRunner)
+@InjectWith(XtextUiInjectorProvider)
+class XtextFoldingTest extends AbstractFoldingTest {
```

Comment:
is it possible to write this in java directly

Reply:
Because of the heavy usage of multi-line strings I think Xtend is a better choice here.
How would you replace the Xtend multi-line strings in Java?

Reply:
something like Xtext2EcoreTransformerTest?

Reply:
> 
> 
> something like Xtext2EcoreTransformerTest?

I do not know if it is a good idea to use hard-coded linux line endings as currently in the [Xtext2EcoreTransformerTest](https://github.com/eclipse/xtext-extras/blob/6c5a0cf793fc5aee7ab6248b9fb17c22bbbd0fcb/org.eclipse.xtext.extras.tests/src/org/eclipse/xtext/xtext/ecoreInference/Xtext2EcoreTransformerTest.java#L126-L132).

Reply:
Working with fixed line delimiters is usually better than monkey-patching everything to system specific line delimiters back and forth.
We should make sure though that we do have tests for windows and unix line endings in place.